GVFS/GVFS.Common/Maintenance/PackfileMaintenanceStep.cs

```diff
@@ -150,7 +138,23 @@ protected override void PerformMaintenance()
                     return;
                 }
 
-                GitProcess.Result expireResult = this.RunGitCommand((process) => process.MultiPackIndexExpire(this.Context.Enlistment.GitObjectsRoot), nameof(GitProcess.MultiPackIndexExpire));
+                try
+                {
+                    // If a LibGit2Repo is active, then it may hold handles to the .idx and .pack files we want
+                    // to delete during the 'git multi-pack-index expire' step. If one starts during the step,
+                    // then it can still block those deletions, but we will clean them up in the next run. By
+                    // running CloseActiveRepos() here, we ensure that we do not run twice with the same
+                    // LibGit2Repo active across two calls. A "new" repo should not hold handles to .idx files
+                    // that do not have corresponding .pack files, so we will clean them up in CleanStaleIdxFiles().
+                    this.Context.Repository.CloseActiveRepo();
+
+                    GitProcess.Result expireResult = this.RunGitCommand((process) => process.MultiPackIndexExpire(this.Context.Enlistment.GitObjectsRoot), nameof(GitProcess.MultiPackIndexExpire));
+                }
+                finally
+                {
+                    this.Context.Repository.OpenRepo();

```

Comment:
Moving review to this PR (from #1279) as I see some other comments have been posted on this one.

Should this `finally` block run even if the exception thrown is a `StoppingException`?  I wasn't sure that we need to hit:

```
            // Run a test on the shared repo to ensure the object store
            // is loaded, as that is what takes a long time with many packs.
            this.GetSharedRepo().ObjectExists(GVFSConstants.AllZeroSha);
```

during shutdown.

(Note: I'm not sure if there are any other exception types that might be thrown from `RunGitCommand`)

Reply:
I never _expect_ other exceptions, but this is a performance optimization, so skipping the `OpenRepo()` call doesn't break us.

Reply:
So I'll remove the `try`/`finally`.